tests/test_core.py

```diff
@@ -1118,6 +1118,40 @@ class MyClass(NSObject):
             class MyClass(NSObject):  # noqa: F811
                 pass
 
+    def test_class_auto_rename(self):
+        """Test automatic renaming system of ObjCClass."""
+
+        ObjCClass.auto_rename = True
+
+        for _ in range(2):
+
+            class TestClass2(NSObject):
+                pass
+
+        ObjCClass.auto_rename = False
+
+        for _ in range(2):
+
+            class TestClass(NSObject, auto_rename=True):
+                pass
+
+    def test_protocol_auto_rename(self):
+        """Test automatic renaming system of ObjCProtocol."""
+
+        ObjCProtocol.auto_rename = True
+
+        for _ in range(2):
+
+            class TestProtocol(metaclass=ObjCProtocol):
+                pass
+
+        ObjCProtocol.auto_rename = False
+
+        for _ in range(2):
+
+            class TestProtocol2(metaclass=ObjCProtocol, auto_rename=True):
+                pass
```

Comment:
As above, this verifies there's no error raised, but doesn't verify any behavior.